src/main/java/io/openliberty/tools/intellij/lsp4mp/lsp4ij/server/ProcessStreamConnectionProvider.java

```diff
@@ -160,5 +169,44 @@ public String toString() {
         return "ProcessStreamConnectionProvider [commands=" + this.getCommands() + ", workingDir=" //$NON-NLS-1$//$NON-NLS-2$
                 + this.getWorkingDirectory() + "]"; //$NON-NLS-1$
     }
+    public boolean checkJavaHome(String javaHome, String serverType) {
```

Comment:
Now that this method returns a boolean, it is good practice to name the method in the style of "isCondition". So, in this case, I would recommend naming the method `isJaveHomeValid`. And we would want to return `true` when Java home is valid, so we would need to flip the return values around (ie. return `false` for any of the negative cases where Java home is not valid and return `true` at the end of the method for the positive case). This means you will also need to flip the method calls from the other files (ie. `if !isJavaHomeValid() { return; }`)